Sources/BrowserServicesKit/SecureVault/AutofillSecureVault.swift

```diff
@@ -61,37 +75,28 @@ public protocol SecureVault {
     func deleteCreditCardFor(cardId: Int64) throws
 }
 
-/// Protocols can't be nested, but classes can.  This struct provides a 'namespace' for the default implementations of the providers to keep it clean for other things going on in this library.
-internal struct SecureVaultProviders {
-
-    var crypto: SecureVaultCryptoProvider
-    var database: SecureVaultDatabaseProvider
-    var keystore: SecureVaultKeyStoreProvider
-
-}
+public class DefaultAutofillSecureVault<T: AutofillDatabaseProvider>: AutofillSecureVault {
```

Comment:
I know we talked about it and know the reason why, but something seems off to me if the concrete implementation is itself generic. Lets have a wee chat later.